source/sundials/ida_interface.cc

```diff
@@ -0,0 +1,602 @@
+                      "    algebraic components in the function get_differential_components.\n"
+                      " use_y_dot: compute all components of y, given y_dot.",
+                      Patterns::Selection("none|use_y_diff|use_y_dot"));
+
+    prm.add_parameter("Initial condition Newton parameter", ic_alpha);
+
+    prm.add_parameter("Initial condition Newton max iterations", ic_max_iter);
+
+    prm.add_parameter("Use local tolerances", use_local_tolerances);
+
+    prm.add_parameter("Show output of time steps", verbose);
+  }
+
+
+  template <typename VectorType>
+  unsigned int IDAInterface<VectorType>::solve_dae(VectorType &solution,
+                                                   VectorType &solution_dot)
+  {
+
+    unsigned int system_size = solution.size();
+    unsigned int local_system_size = system_size;
+
+    double t = initial_time;
+    double h = initial_step_size;
+    unsigned int step_number = 0;
+
+    int status;
+
+    // The solution is stored in
+    // solution. Here we take only a
+    // view of it.
+#ifdef DEAL_II_WITH_MPI
+    if (is_serial_vector<VectorType>::value == false)
+      {
+        IndexSet is = solution.locally_owned_elements();
+        local_system_size = is.n_elements();
+
+        yy        = N_VNew_Parallel(communicator,
+                                    local_system_size,
+                                    system_size);
+
+        yp        = N_VNew_Parallel(communicator,
+                                    local_system_size,
+                                    system_size);
+
+        diff_id   = N_VNew_Parallel(communicator,
+                                    local_system_size,
+                                    system_size);
+
+        abs_tolls = N_VNew_Parallel(communicator,
+                                    local_system_size,
+                                    system_size);
+      }
+    else
+#endif
+      {
+        Assert(is_serial_vector<VectorType>::value,
+               ExcInternalError("Trying to use a serial code with a parallel vector."));
+        yy        = N_VNew_Serial(system_size);
+        yp        = N_VNew_Serial(system_size);
+        diff_id   = N_VNew_Serial(system_size);
+        abs_tolls = N_VNew_Serial(system_size);
+      }
+    reset_dae(initial_time,
+              solution,
+              solution_dot,
+              initial_step_size,
+              true);
+
+    double next_time = initial_time;
+
+    output_step( 0, solution, solution_dot, 0);
+
+    while (t<final_time)
+      {
+
+        next_time += output_period;
+        if (verbose)
+          {
+            pcout << " "//"\r"
+                  << std::setw(5) << t << " ----> "
+                  << std::setw(5) << next_time
+                  << std::endl;
+          }
+        status = IDASolve(ida_mem, next_time, &t, yy, yp, IDA_NORMAL);
+
+        status = IDAGetLastStep(ida_mem, &h);
+        AssertThrow(status == 0, ExcMessage("Error in IDA Solver"));
```

Comment:
I think it would be nice to output the error code. I think there is already something like `AssertMPI` or `Assert(PETSc)` that you could steal for here.